Api/Data/AmwalButtonConfigInterface.php

```diff
@@ -335,4 +337,17 @@ public function getCartId(): ?string;
      */
     public function setCartId(?string $cartId): AmwalButtonConfigInterface;
 
+
+    /**
+     * @return string|null
+     */
+    public function getQuote(): string;
+
+    /**
+     * Set quote data.
+     *
+     * @param string $quote
+     * @return AmwalButtonConfigInterface
+     */
+    public function setQuote($quote): AmwalButtonConfigInterface;
```

Comment:
I believe the type hints here are incorrect since we're setting this with `$quote->getData()`. Perhaps we can use `$quote->toArray()` and then set the type hints like so:

```suggestion
    /**
     * @return array|null
     */
    public function getQuote(): ?array;

    /**
     * Set quote data.
     *
     * @param array $quote
     * @return AmwalButtonConfigInterface
     */
    public function setQuote(array $quote): AmwalButtonConfigInterface;
```